### PR TITLE
clippy: Rename `RequestId::new()` to `RequestId::next()` and fix one more issue in `servo/lib.rs`

### PR DESCRIPTION
**Pull request by komutumu**
This PR addresses clippy warnings in files; 
- components/servo/lib.rs 
- components/shared/net/request.rs
- components/script/fetch.rs

---
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes are part of  #31500
- [X] These changes do not require tests because they do not touch functionality.

### CODE REVIEW

File: components/servo/lib.rs [hunk 1]
```diff
@@ -228,6 +228,7 @@ where
         feature = "tracing",
         tracing::instrument(skip(embedder, window), fields(servo_profiling = true))
     )]
+    #[allow(clippy::new_ret_no_self)]
```
Comment (mrobinson) [inline components/servo/lib.rs:231]:
```suggestion

    #[allow(clippy::new_ret_no_self)]
```

File: components/shared/net/request.rs [hunk 2]
```diff
@@ -22,8 +22,8 @@ use crate::{ReferrerPolicy, ResourceTimingType};
 /// An id to differeniate one network request from another.
 pub struct RequestId(usize);
 
-impl RequestId {
-    pub fn new() -> Self {
+impl Default for RequestId {
+    fn default() -> Self {
```
Comment (mrobinson) [inline components/shared/net/request.rs:26]: `default()` doesn't quite capture what this is doing. Instead please call this `next()`:

```suggestion
impl RequestId {
    fn next() -> Self {
```

**Review by mrobinson — CHANGES_REQUESTED** [review]

File: components/shared/net/request.rs (hunk 2 above)
Comment (jahielkomu) [reply components/shared/net/request.rs:26]: Okay Sir, let me do that.

File: components/servo/lib.rs (hunk 1 above)
Comment (jahielkomu) [reply components/servo/lib.rs:231]: @mrobinson do you suggest I add an empty line before the ```allow``` here?

Comment (jahielkomu) [reply components/servo/lib.rs:231]: When I add the empty line, ```test-tidy``` reports some issues from  there.

**Comment by simonwuelker** [comment]
> These changes fix https://github.com/servo/servo/issues/31500

FYI, if you phrase your PR description like that then github will automatically close #31500 as completed when this PR is merged. Maybe rephrase it to "These changes are part of #31500".

**Comment by jahielkomu** [comment]
> > These changes fix #31500
> 
> FYI, if you phrase your PR description like that then github will automatically close #31500 as completed when this PR is merged. Maybe rephrase it to "These changes are part of #31500".

The part of  "These changes fix __"  is already there on the part where the PRs are created, so I think to largely address it would be from that point.
Thank you, let me edit mine.

File: components/servo/lib.rs (hunk 1 above)
Comment (mrobinson) [reply components/servo/lib.rs:231]: Apologies. My suggestion was a mistake. Please discard it.

File: components/shared/net/request.rs (hunk 2 above)
Comment (jahielkomu) [reply components/shared/net/request.rs:26]: @mrobinson when ```fn next()``` is used ```build -d``` reports errors ``` RequestId` is not an iterator``` 
Should we rename it to something else?

Comment (mrobinson) [reply components/shared/net/request.rs:26]: `next_id` sounds great. Thanks for checking!

Comment (jahielkomu) [reply components/shared/net/request.rs:26]: > `next_id` sounds great. Thanks for checking!

@mrobinson, there appears to still be an issue  with renaming it to anything else but ```default``` unless ```fn default``` is implemented in some way. Let me share a screenshot. 

Screenshot from 2024-10-17 18-31-11

Comment (simonwuelker) [reply components/shared/net/request.rs:26]: you need to remove the `impl Default for RequestId` block and use `impl RequestId` instead.

Comment (jahielkomu) [reply components/shared/net/request.rs:26]: > you need to remove the `impl Default for RequestId` block and use `impl RequestId` instead.

That is the warning we are trying to work on, the only other option would be to ```allow``` it

Comment (mukilan) [reply components/shared/net/request.rs:26]: The warning is triggered only because the method is currently named `new`. It will disappear if we renamed `new` to `next_id`, so there is no need to `allow` it or have `impl Default`.

Comment (jahielkomu) [reply components/shared/net/request.rs:26]: > The warning is triggered only because the method is currently named `new`. It will disappear if we renamed `new` to `next_id`, so there is no need to `allow` it or have `impl Default`.

You are right, looking at the suggestion provided by @mrobinson ```impl Default``` is eliminated. 
Thank you @mukilan

**Review by mrobinson — APPROVED** [review]

**Comment by simonwuelker** [comment]
(PR title is wrong, the method is named `RequestId::next()` not `RequestId::next_id()`)

**Comment by mrobinson** [comment]
> (PR title is wrong, the method is named `RequestId::next()` not `RequestId::next_id()`)

Thanks for pointing that out. I missed that clippy wasn't actually complaining about the `next()` name.
